.github/workflows/run-td-build.yml

```diff
@@ -43,3 +43,25 @@ jobs:
           azure-devops-token: '${{ secrets.AZURE_TOUCHDOWN_BUILD_PIPELINE_SECRET }}'
           # Only commit updated files back when explicitly requested.
           azure-pipeline-variables: "{'commit': ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.commit_strings }} }"
+
+  check_failure:
+    runs-on: ubuntu-latest
+    needs: runTDBuild
+    if: failure()
+    name: Create GitHub issue on failure
+    steps:
+      # Failure of this action indicates a problem with our infrastructure.
+      # Always file a GitHub issue to alert the OCE of the problem.
+      - name: Create GitHub issue
+        id: create-issue
+        run: |
+          # check for an issue that is already open
+          curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/search/issues?q=org:Azure+repo:communication-ui-library+label:TDBUILD_FAILURE+state:open > $HOME/tdbuild_issues.json
+          issue_count=$(jq -r '.total_count' $HOME/tdbuild_issues.json)
+          if [ $issue_count -gt 0 ]; then
+            echo "No need to create new issue, one already exists"
+          else
+            curl -X POST -H 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' https://api.github.com/repos/Azure/communication-ui-library/issues -d '{"title":"Action to trigger TD Build failed", "body": "Please investigate failures of https://github.com/Azure/communication-ui-library/actions/workflows/run-td-build.yml", "labels":["TDBUILD_FAILURE"]}'

```

Comment:
Any way we can link to this build that failed? I know for the main CI fail we don't and it would be really really helpful to put a link to the failed build

Reply:
I got lazy and just linked to the actions page (you see all the latest builds, hopefully including the red ones)